### PR TITLE
autoload experiment_id

### PR DESCRIPTION
**Pull request by alexvpickering**
# Background
#### Link to issue 

https://biomage.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=BIOMAGE&modal=detail&selectedIssue=BIOMAGE-1239

#### Links to any Pull Requests related to this

https://github.com/biomage-ltd/worker/pull/159

#### Anything else the reviewers should know about the changes here

also fixes error tracing issue where sometimes error wasn't reported properly 

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: pipeline-runner/init.R [hunk 1]
```diff
@@ -187,16 +198,29 @@ call_data_processing <- function(task_name, input, pipeline_config) {
...
         message('Count matrix uploaded to ', pipeline_config$processed_bucket, ' with key ',object_key)
+
+        # move scdata to worker (dev only)
+        if (pipeline_config$cluster_env == 'development') {
+            expid_path <- file.path('/worker/data', experiment_id)
+            dir.create(expid_path, showWarnings = FALSE)
+
+            saveRDS(scdata, file.path(expid_path, 'r.rds'))
+            saveRDS(experiment_id, '/worker/data/current_expid.rds')
+        }
```
Comment (alexvpickering) [inline pipeline-runner/init.R:217]: oliver suggested to move save before upload in case upload fails with timeout

Comment (alexvpickering) [reply pipeline-runner/init.R:217]: Oliver mentioned that could also just skip uploading locally all together

File: local-runner/package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "coverage": "jest --coverage --silent",
     "dev": "nodemon src/app.js",
     "build": "docker build --build-arg GITHUB_PAT=${GITHUB_API_TOKEN} -t biomage-pipeline-runner ../pipeline-runner",
-    "start": "DEBUG_PATH=$(pwd)/debug node src/app.js",
+    "start": "WORKER_DIR=$(pwd)/../../worker/data DEBUG_PATH=$(pwd)/debug node src/app.js",
```
Comment (alexvpickering) [inline local-runner/package.json:13]: martin suggested making this user setable

File: pipeline-runner/init.R [hunk 3]
```diff
@@ -179,6 +184,12 @@ call_data_processing <- function(task_name, input, pipeline_config) {
         message("Single-cell data loaded.")
     }
 
+    # update and upload cellSets
+    if (upload_cell_sets) {
```
Comment (ogibson-biomage) [inline pipeline-runner/init.R:188]: Is this fom the fix metadata table changes? Why is it here?

File: pipeline-runner/init.R [hunk 4]
```diff
@@ -187,16 +198,30 @@ call_data_processing <- function(task_name, input, pipeline_config) {
...
+
+        # move scdata to worker (direct for dev only)
+        if (upload_count_matrix & pipeline_config$cluster_env != 'development') {
+            object_key <- upload_matrix_to_s3(pipeline_config, experiment_id, scdata)
+            message('Count matrix uploaded to ', pipeline_config$processed_bucket, ' with key ',object_key)
+
+        } else if (upload_count_matrix) {
+            expid_path <- file.path('/worker/data', experiment_id)
+            dir.create(expid_path, showWarnings = FALSE)
+            saveRDS(scdata, file.path(expid_path, 'r.rds'))
```
Comment (ogibson-biomage) [inline pipeline-runner/init.R:216]: Iva is concerned that we are diverging from our aws environment. If it helps prevent the timeout it's okay, but we need to make sure that it actually helps avoiding the timeout.

Comment (alexvpickering) [reply pipeline-runner/init.R]: Can you test? I don't experience timeouts

**Review by ivababukova — APPROVED** [review]

**Pull request closed without merging**